src/ucp/core/ucp_ep.h

```diff
@@ -306,10 +306,19 @@ typedef struct ucp_ep {
  * Status of protocol-level remote completions
  */
 typedef struct {
-    ucs_queue_head_t              reqs;         /* Queue of flush requests which
-                                                   are waiting for remote completion */
-    uint32_t                      send_sn;      /* Sequence number of sent operations */
-    uint32_t                      cmpl_sn;      /* Sequence number of completions */
+    union {
+        struct {
+            ucs_queue_head_t              reqs;         /* Queue of flush requests which
+                                                           are waiting for remote completion */
+            uint32_t                      send_sn;      /* Sequence number of sent operations */
+            uint32_t                      cmpl_sn;      /* Sequence number of completions */
+        } sw;
```

Comment:
what does sw mean?

Reply:
software flush, part of software emulation layer

Reply:
i think that we need to come up with a better name. sw is very generic.
maybe sw_flush?
anyway, can apply this in the next commit.

Reply:
this is a part of `ucp_ep_flush_state_t`, so I think this is clear enough that this is about flush.
usage looks like `flush_state->sw.reqs)`

Reply:
yeah i saw the usage. sw seems too generic to me...